src/bin/main.ml

```diff
@@ -110,15 +146,22 @@ let start urls sandbox git =
   Log.app (fun l ->
       l "Starting %s %s ..." (Filename.basename Sys.argv.(0)) Version.v
     );
-  begin match git with
-    | None      -> In_memory_store.connect ()
-    | Some path ->
-      let prefix = if sandbox then "." else "" in
-      Git_fs_store.connect (prefix ^ path)
-  end >>= fun make_root ->
-  Lwt_list.iter_p
-    (Datakit_conduit.accept_forever ~make_root ~sandbox ~serviceid)
-    urls
+  let serve_http = match listen_http with
+    | None     -> Lwt.return_unit
+    | Some uri -> http_server (Uri.of_string uri) git
+  in
+  let serve_9p =
+    begin match git with
+      | None      -> In_memory_store.connect ()
+      | Some path ->
+        let prefix = if sandbox then "." else "" in
+        Git_fs_store.connect (prefix ^ path)
+    end >>= fun make_root ->
+    Lwt_list.iter_p
+      (Datakit_conduit.accept_forever ~make_root ~sandbox ~serviceid)
+      listen_9p
+  in
+  Lwt.join [ serve_http; serve_9p ]
```

Comment:
Using `Lwt.join` hides errors, as we discovered before. We should just make a list of the threads we actually want to run and use `Lwt.choose`.